### ISSUE TITLE
2.0-dev webserver config

### ISSUE THREAD

**Issue by jkowalleck** [k4cg/nichtparasoup#115]
we need some kind of config

**Comment by jkowalleck** [comment]
maybe a `yaml` file format like this 

```
## yust some stupid idea ##

webserver:
  # werkzeug config foo
  host: "0.0.0.0"
  port: 5000
  debug: false

server:
  debug: false
  features:
    # maybe add feature switches for stat(u)s, flush, reset etc later ...

nichtparasoup:
  # how many images should be kept in the cache, until a crawl is triggered automatically. find a better word
  crawler_upkeep: 30
  debug: false
  crawlers:
    # list of crawlers and their configs
    # each crawler type has its own config. not sure about type checks.
    #   maybe each crawler implementation will type check its config individually. just pass in what you have...
    - type: Reddit
      weight: 23
      config:
        subredit: awwww
    - type: Reddit
      weight: 42
      config:
        subredit: new
    - type: Picsum
      weright: 13.37
      config:
        width: 800
        height: 600
    - type: Dummy
      weright: 0.0.1
      config:
        image_uri: "https://some.image/wow such pic.gif"


```

**Comment by jkowalleck** [comment]
in preparation, each crawler needs to implement the static method `check_config()` 
this method throws errors if the confic is broken. 
this function will filter/fix the input config and return a config it can work with.

this method is also part of the `__init__()` of each crawler.
this means atleast the constructor will check if a function is broken and prevent further issues.

this basically means, that the config needs not to reallly check on the crawlers config. as long as it is a dictionary everything is fine ...

**Comment by jkowalleck** [comment]
maybe use https://cli.dev/

**Comment by jkowalleck** [comment]
closed via #141
